src/screens/Assistant/SelectFavouritesBottomSheet.tsx

```diff
@@ -0,0 +1,249 @@
+        <ThemeText type="body__primary" style={styles.lineIdentiferText}>
+          {lineIdentifier} {lineName}
+        </ThemeText>
+
+        <ThemeText type="body__secondary" style={styles.secondaryText}>
+          {t(SelectFavouriteDeparturesText.departures.from)} {departureStation}
+        </ThemeText>
+      </View>
+
+      <View>
+        <FixedSwitch
+          accessibilityHint={t(SelectFavouriteDeparturesText.switch.a11yhint)}
+          value={active}
+          onChange={() => handleSwitchFlip(favouriteId)}
+          style={[
+            styles.toggle,
+            Platform.OS === 'android' ? styles.androidToggle : styles.iosToggle,
+          ]}
+        />
+      </View>
+    </View>
+  );
+};
+
+type SelectFavouritesBottomSheetProps = {
+  close: () => void;
+};
+
+const SelectFavouritesBottomSheet = ({
+  close,
+}: SelectFavouritesBottomSheetProps) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const {
+    favoriteDepartures,
+    frontPageFavouriteDepartures,
+    addFrontPageFavouriteDeparture,
+    removeFrontPageFavouriteDeparture,
+  } = useFavorites();
+  const favouriteItems = favoriteDepartures ?? [];
+  const frontpageFavouriteItems = frontPageFavouriteDepartures ?? [];
+
+  const handleSwitchFlip = (favouriteId: string) => {
+    const favouriteInFavourites = favouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    const favouriteInFrontpageFavourites = frontpageFavouriteItems.find(
+      (item) => item.id === favouriteId,
+    );
+
+    if (!favouriteInFrontpageFavourites) {
+      const favourite = favoriteDepartures.find(
+        (departure) => departure.id === favouriteId,
+      );
+      if (!favourite) return;
+      addFrontPageFavouriteDeparture(favourite);
+    } else if (favouriteInFavourites) {
+      removeFrontPageFavouriteDeparture(favouriteInFrontpageFavourites);
+    }
+  };
+
+  return (
+    <BottomSheetContainer>
+      <ScreenHeaderWithoutNavigation
+        title={t(SelectFavouriteDeparturesText.header.text)}
+        leftButton={{
+          type: 'cancel',
+          onPress: close,
+          text: 'Close',
+        }}
+        color="background_1"
+      />
+
+      <ScrollView style={styles.flatListArea}>
+        {favoriteDepartures.length > 0 && (
+          <>
+            <ThemeText style={styles.questionText} type="heading__component">
+              {t(SelectFavouriteDeparturesText.title.text)}
+            </ThemeText>
+
+            <View>
+              {favouriteItems &&
+                favouriteItems.map((departureDetails, index) => {
+                  return (
+                    <View key={departureDetails.id}>
+                      <SelectableFavouriteDeparture
+                        handleSwitchFlip={handleSwitchFlip}
+                        favouriteId={departureDetails.id}
+                        active={
+                          frontPageFavouriteDepartures.find(
```

Comment:
Can be simplified to  
```
  !!frontPageFavouriteDepartures.find(
                              (departure) => departure.id === departureDetails.id,
                          )
```

Reply:
Absolutely, fixing that now. Tried to remember that one, but no success at that time 😅